[Bug]: v7.0.0-beta.43 No loader is configured for ".woff2" files

### Describe the bug


```node
info => Building manager..
✘ [ERROR] No loader is configured for ".woff2" files: ../core/dist/2a8c9ca1.woff2

    ../core/dist/style.css:1:7211:
      1 │ ...;font-family:RalewayVariable;font-style:normal;font-weight:100 900;src:url(2a8c9ca1.woff2) format("woff2");unicode-range:u+0301,u+0400-045f,u+0490-0491,u+04b0-0...
        ╵      
```

```node
ERR! /workspaces/web/node_modules/@storybook/builder-manager/node_modules/esbuild/lib/main.js:1604
ERR!   let error = new Error(`${text}${summary}`);
ERR!               ^
ERR! 
ERR! Error: Build failed with 5 errors:
ERR! ../core/dist/style.css:1:7211: ERROR: No loader is configured for ".woff2" files: ../core/dist/2a8c9ca1.woff2
ERR! ../core/dist/style.css:1:7410: ERROR: No loader is configured for ".woff2" files: ../core/dist/573b44dd.woff2
ERR! ../core/dist/style.css:1:7626: ERROR: No loader is configured for ".woff2" files: ../core/dist/b9eae7ee.woff2
ERR! ../core/dist/style.css:1:7909: ERROR: No loader is configured for ".woff2" files: ../core/dist/1fd62707.woff2
ERR! ../core/dist/style.css:1:8146: ERROR: No loader is configured for ".woff2" files: ../core/dist/23c83bfc.woff2
ERR!     at failureErrorWithLog (/workspaces/web/node_modules/@storybook/builder-manager/node_modules/esbuild/lib/main.js:1604:15)
ERR!     at <anonymous> (/workspaces/web/node_modules/@storybook/builder-manager/node_modules/esbuild/lib/main.js:1056:28)
ERR!     at runOnEndCallbacks (/workspaces/web/node_modules/@storybook/builder-manager/node_modules/esbuild/lib/main.js:1476:61)
ERR!     at buildResponseToResult (/workspaces/web/node_modules/@storybook/builder-manager/node_modules/esbuild/lib/main.js:1054:7)
ERR!     at <anonymous> (/workspaces/web/node_modules/@storybook/builder-manager/node_modules/esbuild/lib/main.js:1166:14)
ERR!     at responseCallbacks.<computed> (/workspaces/web/node_modules/@storybook/builder-manager/node_modules/esbuild/lib/main.js:701:9)
ERR!     at handleIncomingPacket (/workspaces/web/node_modules/@storybook/builder-manager/node_modules/esbuild/lib/main.js:756:9)
ERR!     at Socket.readFromStdout (/workspaces/web/node_modules/@storybook/builder-manager/node_modules/esbuild/lib/main.js:677:7)
ERR!     at Socket.emit (node:events:513:28)
ERR!     at addChunk (node:internal/streams/readable:315:12)
ERR!  /workspaces/web/node_modules/@storybook/builder-manager/node_modules/esbuild/lib/main.js:1604
ERR!   let error = new Error(`${text}${summary}`);
ERR!               ^
```

### To Reproduce

```shell
Testing the migration from 6.5 to 7 beta...
Import a custom stylesheet in manager.js that loads a local woff2 fonts

Run `npx storybook build`
```


### System

```shell
Environment Info:

  System:
    OS: Linux 5.4 Ubuntu 20.04.5 LTS (Focal Fossa)
    CPU: (4) x64 Intel(R) Xeon(R) Platinum 8272CL CPU @ 2.60GHz
  Binaries:
    Node: 16.19.0 - /usr/local/share/nvm/versions/node/v16.19.0/bin/node
    Yarn: 1.22.19 - /usr/bin/yarn
    npm: 8.18.0 - /workspaces/web/node_modules/.bin/npm
  npmPackages:
    @storybook/addons: 7.0.0-beta.43 => 7.0.0-beta.43 
    @storybook/builder-vite: ^0.4.0 => 0.4.0
```


### Additional context

_No response_

--- a/code/lib/builder-manager/src/index.ts
+++ b/code/lib/builder-manager/src/index.ts
@@ -59,6 +59,7 @@ export const getConfig: ManagerBuilder['getConfig'] = async (options) => {
       '.svg': 'dataurl',
       '.webp': 'dataurl',
       '.webm': 'dataurl',
+      '.woff2': 'dataurl',
     },
     target: ['chrome100'],
     platform: 'browser',

